runtime/assembly/models/header.ts

```diff
@@ -30,21 +31,34 @@ export class Header {
      * Field used to store any chain-specific auxiliary data, which could help the light clients interact with the block 
      * without the need of accessing the full storage as well as consensus-related data including the block signature. 
      */
-    public digest: Option<Hash> // TODO not ready..
+    public digests: Option<DigestItem[]>
 
-    constructor(parentHash: Hash, number: CompactInt, stateRoot: Hash, extrinsicsRoot: Hash, digest: Option<Hash>) {
+    constructor(parentHash: Hash, number: CompactInt, stateRoot: Hash, extrinsicsRoot: Hash, digests: Option<DigestItem[]>) {
         this.parentHash = parentHash;
         this.number = number;
         this.stateRoot = stateRoot;
         this.extrinsicsRoot = extrinsicsRoot;
-        this.digest = digest;
+        this.digests = digests;
     }
 
     /**
     * SCALE Encodes the Header into u8[]
     */
+    // TODO fix in another PR
     toU8a(): u8[] {
-        let digest = this.digest.isSome() ? (<Hash>this.digest.unwrap()).toU8a() : Constants.EMPTY_BYTE_ARRAY;
+        let digest:u8[] = [];
+        if (this.digests.isSome()) {
+            let digestItemArray = <DigestItem[]>this.digests.unwrap();
+            const length = new CompactInt(digestItemArray.length);
+            digest.concat(length.toU8a());
```

Comment:
isn't it supposed to be `digest = digest.concat(length.toU8a());`

Reply:
yes, it is. I've found the bug and fixed it in the next PR.

Good catch.